### PR TITLE
Nonce Manager

### PR DESCRIPTION
**Pull request by MusabShakeel576**
Implement NonceManager to send multiple independent UserOp batch transactions at once.

### CODE REVIEW

File: src/utils/nonceManager.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// NonceManager handle independent non-blocking UserOp transactions
+export class NonceManager {
+  private _nonceKey: number = -1;
+
+  constructor() {}
+
+  increment = () => {
```
Comment (leonprou) [inline src/utils/nonceManager.ts:7]: where do you call incerement?

File: src/sdk.ts [hunk 2]
```diff
@@ -86,12 +98,24 @@ export class FuseSDK {
...
     calls: Array<ICall>,
-    txOptions?: typeof Variables.DEFAULT_TX_OPTIONS
+    isIndependentTransaction?: boolean,
+    txOptions?: typeof Variables.DEFAULT_TX_OPTIONS,
   ): Promise<ISendUserOperationResponse | null | undefined> {
     txOptions = txOptions ?? Variables.DEFAULT_TX_OPTIONS;
     const initialFees = BigInt(txOptions.feePerGas);
     this.setWalletFees(initialFees);
 
+    if(isIndependentTransaction) {
```
Comment (leonprou) [inline src/sdk.ts]: Can you please expain how and when this flag should be used?

Comment (MusabShakeel576) [reply src/sdk.ts]: @leonprou when dev need to execute multiple batches, they can pass this flag to the executeBatch function.

Comment (leonprou) [reply src/sdk.ts]: Ok I see,
I have concerns about the re-initialization of the wallet. It creates a new client so if we use await with the previous client we are loosing the reference.

**Review by LiorAgnin — APPROVED** [review]
LGTM!

**Comment by LiorAgnin** [comment]
@leonprou @MusabShakeel576 Can we merge it?

**Pull request merged**